src/Components/App/App.js

```diff
@@ -18,36 +19,56 @@ const App = () => {
       } catch (error) {
         console.log(error);
       }
-    }
+    };
     fetchAllPokemon();
   }, []);
 
   const get4RdmPokemon = () => {
-    const pokemons = [{}, {}, {}, {}]
-    let randomIndexes = []
-    while(randomIndexes.length <= 4) {
+    const pokemons = [{}, {}, {}, {}];
+    let randomIndexes = [];
+    while (randomIndexes.length <= 4) {
       let randNum = Math.floor(Math.random() * allPokemon.length);
       if (randomIndexes.indexOf(randNum) === -1) randomIndexes.push(randNum);
     }
-    return pokemons.map((emptySlot,i) => {
-      const randomIndex = randomIndexes[i]
+    return pokemons.map((emptySlot, i) => {
+      const randomIndex = randomIndexes[i];
       return allPokemon[randomIndex];
-    })
+    });
+  };
+
+  const fetchSinglePokemon = async (url) => {
+    try {
+      const singlePokemon = await getSinglePokemon(url);
+      setSinglePokemon(singlePokemon);
+    } catch (error) {
+      console.log(error);
+    }
   };
 
+
   return (
     <main className="App">
       <Header />
-      {allPokemon.length && <Route
-        exact
-        path="/game"
-        render={() => (
-          <Game get4RdmPokemon={get4RdmPokemon} pokemons={get4RdmPokemon()}/>
-        )}
-      />}
-      <Route path='/pokedex' render={() => <Pokedex allPokemon={allPokemon}/>}/>
+      <AppContext.Provider value= {{singlePokemon, setSinglePokemon}}>
+        {allPokemon.length && <Route
+          exact
+          path="/game"
+          render={() => (
+            <Game 
+              get4RdmPokemon={get4RdmPokemon} 
+              fetchSinglePokemon={fetchSinglePokemon}/>
+          )}
+        />}
+        <Route path='/pokedex' 
+          render={() => 
+            <Pokedex 
+              allPokemon={allPokemon} 
+              fetchSinglePokemon={fetchSinglePokemon}
```

Comment:
It would be nice to standardize the styling across these views -- the gameboy style is fun on the Pokedex.